src/formatDate.js

```diff
@@ -8,7 +8,38 @@
  * @returns {string}
  */
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
-}
+  const partsDate = date.split(fromFormat[3]);
```

Comment:
The code assumes that the separator is always at index 3 in the `fromFormat` array. This assumption may not hold true for all date formats. Consider dynamically determining the separator based on the format string.